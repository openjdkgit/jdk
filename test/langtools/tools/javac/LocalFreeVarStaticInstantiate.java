/*
 * @test /nodynamiccopyright/
 * @bug 8322882
 * @summary Disallow attempts to access a free variable proxy field from a static method
 * @compile/fail/ref=LocalFreeVarStaticInstantiate.out -XDrawDiagnostics LocalFreeVarStaticInstantiate.java
 */

class LocalFreeVarStaticInstantiate {

    static void foo(Object there) {
        class Local {
            {
                there.hashCode();
            }
            static {
                new Local();    // can't get there from here
            }
        }
    }
}
