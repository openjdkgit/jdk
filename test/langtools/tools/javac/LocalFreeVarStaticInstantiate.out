LocalFreeVarStaticInstantiate.java:16:17: compiler.err.non-static.cant.be.ref: kindname.method, foo(java.lang.Object)
1 error
